packages/stack/src/views/Stack/CardStack.tsx

```diff
@@ -334,6 +335,55 @@ export default class CardStack extends React.Component<Props, State> {
     return state.routes[state.index];
   };
 
+  private getSomeFloatHeaderNeedsAbsolutePositioning() {
+    if (this.props.headerMode !== 'float') {
+      return false;
+    }
+
+    const {
+      closingRouteKeys,
+      openingRouteKeys,
+      replacingRouteKeys,
+    } = this.props;
+    const focusedRoute = this.getFocusedRoute();
+    const visibleKeys = new Set([
+      ...closingRouteKeys,
+      ...replacingRouteKeys,
+      focusedRoute.key,
+    ]);
+    const prevFocusedRoute = this.props.getPreviousRoute({
+      route: focusedRoute,
+    });
+    if (prevFocusedRoute) {
+      visibleKeys.add(prevFocusedRoute.key);
+    }
+    if (openingRouteKeys.length > 0) {
+      for (let scene of this.state.scenes) {
+        const { route } = scene;
+        if (!openingRouteKeys.includes(route.key)) {
+          continue;
+        }
+        const prevRoute = this.props.getPreviousRoute({ route });
+        if (prevRoute) {
+          visibleKeys.add(prevRoute.key);
+        }
+      }
+    }
+
+    for (let scene of this.state.scenes) {
+      const { route, descriptor } = scene;
+      if (!visibleKeys.has(route.key)) {
```

Comment:
Maybe it's simpler to just check if the route is one of the last 2 scenes since the last 2 are always the screens that can be visible at a time. Or check last 2 scenes when closingRouteKeys/openingRouteKeys are non empty, otherwise only the last one. Though the simpler the better. Currently this method looks a bit complex.

Reply:
You're right, this is way more complex than it needs to be. I did some testing and the last two scenes should be all we need to check.